Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -104,42 +121,80 @@ class InAppMessageViewTest: UnitTest {
         XCTAssertNil(view)
     }
 
-    func test_givenAlreadyShowingInAppMessage_whenNewMessageFetched_expectDoNotReplaceContents() {
-        let givenOldInlineMessage = Message.randomInline
-        queueMock.getInlineMessagesReturnValue = [givenOldInlineMessage]
+    @MainActor
+    func test_givenAlreadyShowingMessage_whenSameMessageFetched_expectDoNotReloadTheMessageAgain() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
 
-        let inlineView = InAppMessageView(elementId: givenOldInlineMessage.elementId!)
-        let webViewBeforeFetch = getInAppMessageWebView(fromInlineView: inlineView)
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
 
-        // Make sure message is unique, but has same elementId.
-        let givenNewInlineMessage = Message(messageId: .random, campaignId: .random, elementId: givenOldInlineMessage.elementId)
+        let webViewBeforeFetch = getInAppMessageWebView(fromInlineView: inlineView)
 
-        simulateSdkFetchedMessages([givenNewInlineMessage])
+        await simulateSdkFetchedMessages([givenInlineMessage])
 
         let webViewAfterFetch = getInAppMessageWebView(fromInlineView: inlineView)
 
-        // If the WebViews are different, it means the message was reloaded.
+        // If the WebViews are the same instance, it means the message was not reloaded.
```

Comment:
Since the single-source-of-truth is the UI, I like how this test does an assertion on the View layer. I don't know if the test would increase in accuracy by doing this check. Can you think of a reason to?